R/XcmsExperiment-functions.R

```diff
@@ -1298,7 +1298,7 @@ XcmsExperiment <- function() {
                             processing = paste0("Data converted [", date(),"]"),
                             files = fileNames(from),
                             smoothed = NA)
-    n@phenoData <- new("NAnnotatedDataFrame", as.data.frame(sampleData(from)))
+    n@phenoData <- AnnotatedDataFrame(as.data.frame(sampleData(from)))
```

Comment:
using `new()` vs. directly the constructor, is that a style thing, or are these different class systems ?

Reply:
Creating new S4 objects using a constructor function is a more elegant, and safer way than using `new()`. It's more user friendly and allows also to check and eventually fix/format the input parameters. For the `MSnbase::NAnnotatedDataFrame` there was never a constructor function - in contrast to the `Biobase::AnnotatedDataFrame()`.